src/types.jl

```diff
@@ -821,6 +844,14 @@ Base.hash(p::Pow, u::UInt) = hash(p.exp, hash(p.base, u))
 
 Base.isequal(p::Pow, b::Pow) = isequal(p.base, b.base) && isequal(p.exp, b.exp)
 
+Base.isequal(b::Mul, p::Pow) = isequal(p, b)
+function Base.isequal(p::Pow, b::Mul)
```

Comment:
So does this, because `x^2 * 1` is just Pow `x^2`.

Reply:
we should never construct `x^2 * 1`, always go directly to `x^2`.